src/proxyProvider.ts

```diff
@@ -1,42 +1,41 @@
-import HttpsProxyAgent from "https-proxy-agent/dist/agent";
-import * as url from "url";
+import {
+  HttpsProxyAgent,
+  HttpsProxyAgentOptions,
+} from "https-proxy-agent/dist";
+import { URL } from "url";
 import { workspace } from "vscode";
+import tabnineExtensionProperties from "./globals/tabnineExtensionProperties";
 
-type ProxyAgentSettings = {
-  agent: HttpsProxyAgent | undefined;
-  rejectUnauthorized: boolean;
+type ProxyAgentOptions = {
+  ignoreCertificateErrors?: boolean;
+  ca: Buffer | undefined;
 };
-export default function getHttpsProxyAgent(): ProxyAgentSettings {
-  const proxySettings = getProxySettings();
 
-  if (!proxySettings) {
-    return { agent: undefined, rejectUnauthorized: false };
-  }
+export default function getHttpsProxyAgent(
+  options: ProxyAgentOptions
+): HttpsProxyAgent | undefined {
+  const proxySettings = getProxySettings();
 
-  const proxyUrl = url.parse(proxySettings);
-  if (proxyUrl.protocol !== "https:" && proxyUrl.protocol !== "http:") {
-    return { agent: undefined, rejectUnauthorized: false };
+  if (!proxySettings || !tabnineExtensionProperties.useProxySupport) {
+    return undefined;
   }
 
-  const rejectUnauthorized = workspace
-    .getConfiguration()
-    .get("http.proxyStrictSSL", true);
-  const parsedPort: number | undefined = proxyUrl.port
-    ? parseInt(proxyUrl.port, 10)
-    : undefined;
-  const port = Number.isNaN(parsedPort) ? undefined : parsedPort;
+  const proxyUrl = new URL(proxySettings);
 
-  const proxyOptions = {
-    host: proxyUrl.hostname,
-    port,
-    auth: proxyUrl.auth,
-    rejectUnauthorized,
+  const proxyOptions: HttpsProxyAgentOptions = {
```

Comment:
```suggestion
  const proxyOptions: HttpsProxyAgentOptions = {...proxyUrl, ca: options.ca, rejectUnauthorized: !options.ignoreCertificateErrors
```

Reply:
I tried it initially and it didn't work, not sure why .. you can try this in the browser console:
`{...new URL(window.location.href)}`

Reply:
Apparently URL properties are not enumerable https://stackoverflow.com/questions/48232131/why-does-object-assign-not-copy-the-properties-of-a-url-object